### PR TITLE
adds a setting for sd cancel command

### PR DESCRIPTION
**Pull request by jneilliii**
not binding to settingsViewModel as expected, needs adjustment to API?

  * [x] Your changes are not possible to do through a plugin and relevant
    to a large audience (ideally all users of OctoPrint)
  * [x] If your changes are large or otherwise disruptive: You have
    made sure your changes don't interfere with current development by
    talking it through with the maintainers, e.g. through a
    Brainstorming ticket
  * [x] Your PR targets OctoPrint's devel branch if it's a completely
    new feature, or maintenance if it's a bug fix or improvement of
    existing functionality for the current stable version (no PRs
    against master or anything else please)
  * [x] Your PR was opened from a custom branch on your repository
    (no PRs from your version of master, maintenance or devel please),
    e.g. dev/my_new_feature or fix/my_bugfix
  * [x] Your PR only contains relevant changes: no unrelated files,
    no dead code, ideally only one commit - rebase and squash your PR
    if necessary!
  * [x] Your changes follow the existing coding style
  * [x] If your changes include style sheets: You have modified the
    .less source files, not the .css files (those are generated with
    lessc)
  * [x] You have tested your changes (please state how!) - ideally you
    have added unit tests
  * [ ] You have run the existing unit tests against your changes and
    nothing broke
  * [x] You have added yourself to the AUTHORS.md file :)

#### What does this PR do and why is it necessary?
adds a setting for sd cancel command

#### How was it tested? How can it be tested by the reviewer?
installed locally in dev environment and verified new setting in settings

#### What are the relevant tickets if any?
#4368 

#### Screenshots (if appropriate)
image

#### Further notes
~~setting is not binding on initial load for some reason, and not sure why~~

### CODE REVIEW

**Comment by jneilliii** [comment]
> For the settings to bind, it needs to be added to the API output for settings. Unlike with plugins where it is everything minus exclusions, core works with explicitly forming the settings API response.

That's kind of what I thought and noted in my commit message, jut haven't got around to adjusting that part yet.

**Comment by jneilliii** [comment]
Fixed the binding issue, thanks for the pointer Charlie, should be ready to review now. Need to verify it actually sends the command, as the virtual printer is too fast to catch a pause with and I never print from SD, but the logic should be right.

File: src/octoprint/util/comm.py [hunk 1]
```diff
@@ -1752,7 +1752,7 @@ def _on_M400_sent():
             if self.isSdFileSelected():
                 if not external_sd:
                     self.sendCommand(
-                        "M25",
+                        settings().get(["serial", "sdCancelCommand"]),
```
Comment (foosel) [inline src/octoprint/util/comm.py:1755]: This should be read into a variable once at instantiation of the comm instance. See how `self._hello_command` is done.

**Review by foosel — CHANGES_REQUESTED** [review]

File: src/octoprint/util/comm.py (hunk 1 above)
Comment (jneilliii) [reply src/octoprint/util/comm.py]: requested adjustments have been made

**Review by foosel — APPROVED** [review]

**Pull request merged**